### ISSUE TITLE
Expired certificate for https://doc.qgis.org/

### ISSUE THREAD

**Issue by agiudiceandrea** [qgis/QGIS-Website#499]
### URL

https://doc.qgis.org/

### Summary

https://doc.qgis.org/ (instead of https://docs.qgis.org/) appears as the first result in some Google searches e.g. https://www.google.com/search?q=qgis+processing+coveragevalidate
The certificate for https://doc.qgis.org/ expired on August 2024.

See https://lists.osgeo.org/pipermail/qgis-user/2024-December/055040.html.

**Comment by Xpirix** [comment]
Hi @agiudiceandrea , thanks for reporting this.

Hi @timlinux, I think we should also check this later along with others once we have the migration sorted out.

**Comment by agiudiceandrea** [comment]
Should it be possible to just redirect doc.qgis.org to docs.qgis.org?

**Comment by Xpirix** [comment]
> Should it be possible to just redirect doc.qgis.org to docs.qgis.org?

I think both `doc.qgis.org` and `docs.qgis.org` are redirected to the same page, which is https://www.qgis.org/resources/hub/.

However, the domain `doc.qgis.org` probably points to another server where the certificate is not maintained anymore and then redirected to http://www.qgis.org/en/docs/index.html which goes finally to  https://www.qgis.org/resources/hub/ (See the curl result below).

So we think it's more careful to check the server and the domain itself (if it's been used somewhere else) before updating it.

```
$ curl -k https://doc.qgis.org
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>302 Found</title>
</head><body>
<h1>Found</h1>
<p>The document has moved <a href="http://www.qgis.org/en/docs/index.html">here</a>.</p>
<hr>
<address>Apache/2.4.54 (Debian) Server at doc.qgis.org Port 443</address>
</body></html>
```

**Comment by m-kuhn** [comment]
Did we find out which server this is?

**Comment by agiudiceandrea** [comment]
@Xpirix, when an URL like https://doc.qgis.org/3.34/en/docs/user_manual/index.html is used, then there is no redirection.
The IP address from Italy is 195.201.96.242, actually Cloudflare.

**Comment by timlinux** [comment]
As discussed today:

@Xpirix to add rewrite rules on qgis.org

@timlinux to update dns for doc.qgis.org and documentation.qgis.org to point to 95.217.26.231 (our floating IP for QGIS.org) and the redirect rules will take you down to the docs page on the main site.

**Comment by agiudiceandrea** [comment]
Please see also https://github.com/qgis/QGIS-Website/issues/421.

**Comment by agiudiceandrea** [comment]
Now it looks like that:

- URLs like https://doc.qgis.org/3.34/en/docs/user_manual/index.html or https://docs.qgis.org/3.34/en/docs/user_manual/index.html or https://documentation.qgis.org/3.34/en/docs/user_manual/index.html correctly show the documentation pages without any redirection

- the URLs https://doc.qgis.org and https://docs.qgis.org and https://documentation.qgis.org redirect (302) to http://www.qgis.org/en/docs/index.html which in turn redirects (307) to https://www.qgis.org/en/docs/index.html which in turn redirects (301) to https://www.qgis.org/resources/hub which in turn redirects (301) to https://www.qgis.org/resources/hub/

**Comment by Xpirix** [comment]
@agiudiceandrea Yes, that's right and I think these are expected for now according to the status below:

## Current status
- We've redirected `doc.qgis.org` to `docs.qgis.org`
- However, the base URL `docs.qgis.org` and `documentation.qgis.org` (without any path) are redirected to http://www.qgis.org/en/docs/index.html and then to https://www.qgis.org/resources/hub. These rules are defined respectively on the docs server and the QGIS.org server. I think the reason of the base URL redirection is to make the URL https://www.qgis.org/resources/hub the entrypoint of all the docs.
- The docs URL with path (https://docs.qgis.org/3.34/ for example) doesn't have any redirection defined on its server

## Migration Plan
Cc @m-kuhn , @timlinux 
Also replying  #421 and #339 

- Currently, there are some resources along the documentation and the old site (www2.qgis.org) that are sharing the same server
- We will port the build logic of the docs website to the main website, and migrate to a declarative, GitOps model based on CI/CD, ensuring a modern, efficient infrastructure. 
- The old servers, domains, and repositories will be turned off once the migration is complete.

@timlinux Perhaps we should retry to renew the certificates for https://www2.qgis.org (or eventually redirect it) for now.

**Comment by m-kuhn** [comment]
Thanks a lot for the fix @Xpirix !
If I could wish I'd redirect docs.qgis.org the same way as docs.qgis.org/latest does

**Comment by Xpirix** [comment]
My pleasure @m-kuhn , actually, it was mainly @timlinux who was in the action.

> If I could wish I'd redirect docs.qgis.org the same way as docs.qgis.org/latest does

Do you mean redirect docs.qgis.org to https://docs.qgis.org/3.34/en/docs/ instead of https://www.qgis.org/resources/hub/?
It should be possible (should be done from the docs server).

**Comment by m-kuhn** [comment]
Exactly, and in the close future to 3.40.

**Comment by Xpirix** [comment]
All right, I will create an issue for that.
